code/online_challenges/src/codechef/COINS/coins.cpp

```diff
@@ -0,0 +1,34 @@
+#include<bits/stdc++.h>
+using namespace std ;
+#define lli long long int
+unordered_map<lli,lli> mpp;
+lli check(lli n)
+{
+
+  if(n==0)
+    {  mpp[0]=0;
+      return 0;
+    }
+  if(mpp[n]!=0)
+      return mpp[n];
+  else
+      {
+        mpp[n]=check(n/2)+check(n/3)+check(n/4);
+        mpp[n]=max(n,mpp[n]);
+        return mpp[n];
+      }
+}
+int main()
+{
+
+ 	lli  n;
+ 	while(cin>>n)
```

Comment:
Can use `for (lli n; std::cin >> n;)`.

Reply:
OK sir I will change it too

Reply:
Done sir please review changes.

Reply:
and sir I have also added code for implementation of stack using linked list , its not shown as another PR ,please guide me what to do now